### PR TITLE
Fix IDE0044: MakeFieldReadonly part 11

### PR DESCRIPTION
**Pull request by xtqqczze**
https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0044

_Split from #13880 to ease review._

`src\System.Management.Automation\engine\remoting\`

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs [hunk 1]
```diff
@@ -18,12 +18,12 @@ internal class ClientMethodExecutor
...
         /// Transport manager.
         /// </summary>
-        private BaseClientTransportManager _transportManager;
+        private readonly BaseClientTransportManager _transportManager;
 
         /// <summary>
         /// Client host.
         /// </summary>
-        private PSHost _clientHost;
+        private readonly PSHost _clientHost;
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs:26]: Why not make Guid fields readonly as well?

File: src/System.Management.Automation/engine/remoting/client/Job.cs [hunk 2]
```diff
@@ -3824,7 +3824,7 @@ private void HandleRunspaceAvailabilityChanged(object sender, RunspaceAvailabili
         #region Private Members
 
         // helper associated with this job object
-        private RemotePipeline _remotePipeline = null;
+        private readonly RemotePipeline _remotePipeline = null;
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/client/Job.cs:3827]: We can remove the null assignment as it is unneeded.

File: src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs [hunk 3]
```diff
@@ -146,7 +146,7 @@ internal string GetMessage(string resourceString, params object[] args)
 
         #region Private Members
 
-        private static string s_LOCALHOST = "localhost";
+        private static readonly string s_LOCALHOST = "localhost";
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs:149]: This should be `const string`

File: src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs [hunk 4]
```diff
@@ -1286,13 +1286,13 @@ private void WriteErrorCreateRemoteRunspaceFailed(Exception e, Uri uri)
...
-        private List<RemoteRunspace> _toDispose = new List<RemoteRunspace>();
+        private readonly List<RemoteRunspace> _toDispose = new List<RemoteRunspace>();
 
         // List of runspace connect operations.  Need to keep for cleanup.
-        private Collection<List<IThrottleOperation>> _allOperations = new Collection<List<IThrottleOperation>>();
+        private readonly Collection<List<IThrottleOperation>> _allOperations = new Collection<List<IThrottleOperation>>();
 
         // Default FQEID.
-        private string _defaultFQEID = "PSSessionOpenFailed";
+        private readonly string _defaultFQEID = "PSSessionOpenFailed";
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs:1295]: Should be `const string`

File: src/System.Management.Automation/engine/remoting/common/throttlemanager.cs [hunk 5]
```diff
@@ -599,34 +599,34 @@ private void RaiseThrottleManagerEvents()
         /// Default throttle limit - the maximum number of operations
         /// to be processed at a time.
         /// </summary>
-        private static int s_DEFAULT_THROTTLE_LIMIT = 32;
+        private static readonly int s_DEFAULT_THROTTLE_LIMIT = 32;
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/common/throttlemanager.cs:602]: Should be `const int`

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/remoting/client/Job.cs (hunk 2 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/remoting/client/Job.cs:3827]: We could do this in separate PR, clean all code base and force a rule for this.

**Comment by iSazonov** [comment]
@xtqqczze Please look @PaulHigin's comments.

File: src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs (hunk 3 above)
Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs:149]: These changes will be made in follow-up PR, see tracking issue #13914

File: src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs (hunk 4 above)
Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs:1295]: in follow-up PR

Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs:1295]: in follow-up PR

File: src/System.Management.Automation/engine/remoting/common/throttlemanager.cs (hunk 5 above)
Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/common/throttlemanager.cs:602]: in follow-up PR

File: src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs (hunk 1 above)
Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs:26]: Maybe the IDE0044 analyzer doesn't suggest fixes for struct types, I will make changes a new PR

**Comment by iSazonov** [comment]
@PaulHigin Please update your review.

File: src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs (hunk 3 above)
Comment (PaulHigin) [reply src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs:149]: No, please make these changes here.  The Issue can be used to pick up the rest.

File: src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs (hunk 1 above)
Comment (PaulHigin) [reply src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs:26]: No, please make called out changes here.

File: src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs (hunk 4 above)
Comment (PaulHigin) [reply src/System.Management.Automation/engine/remoting/commands/newrunspacecommand.cs:1295]: No, please make called out change.

**Comment by PaulHigin** [comment]
@iSazonov  I'll approve once all requested changes are made.

File: src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs (hunk 1 above)
Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs:26]: I already started work in a new PR, now submitted as #14000.

Comment (PaulHigin) [reply src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs:26]: Ok, that will work.

**Comment by iSazonov** [comment]
@xtqqczze Please look new comments.

File: src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs (hunk 1 above)
Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/client/ClientMethodExecutor.cs:26]: #14000 is ready for review.

**Review by PaulHigin — APPROVED** [review]

**Pull request merged**